### PR TITLE
Fixing layout animations with gap Safari fix

### PR DESCRIPTION
**Pull request by mattgperry**

### CODE REVIEW

File: packages/framer-motion/src/projection/node/create-projection-node.ts [hunk 1]
```diff
@@ -1081,11 +1090,11 @@ export function createProjectionNode<I>({
         /**
          * Animation
          */
-        animationProgress = 0
```
Comment (nvh) [inline packages/framer-motion/src/projection/node/create-projection-node.ts:1084]: What was this used for before?

Comment (mattgperry) [reply packages/framer-motion/src/projection/node/create-projection-node.ts:1084]: Nothing, must have been left around from long ago

File: packages/framer-motion/src/projection/node/create-projection-node.ts [hunk 2]
```diff
@@ -467,6 +467,15 @@ export function createProjectionNode<I>({
 
                             this.startAnimation(animationOptions)
                         } else {
+                            /**
+                             * If the layout hasn't changed and we have an animation that hasn't started yet,
+                             * finish it immediately. Otherwise it will be animating from a location
+                             * that was probably never commited to screen and look like a jumpy box.
+                             */
+                            if (!hasLayoutChanged && this.progress === 0) {
```
Comment (shuangq) [inline packages/framer-motion/src/projection/node/create-projection-node.ts:475]: If the layout hasn't changed, why did it start the animation? 🤔

Comment (mattgperry) [reply packages/framer-motion/src/projection/node/create-projection-node.ts:475]: In a previous render where the layout did change. This is a new render that may have happened so soon that the old animation didn't even start to play.

**Review by shuangq — APPROVED** [review]
Code looks good, any chance we can cover a test? 😬

File: packages/framer-motion/src/projection/node/create-projection-node.ts [hunk 3]
```diff
@@ -1081,11 +1090,11 @@ export function createProjectionNode<I>({
         /**
          * Animation
          */
-        animationProgress = 0
         animationValues?: ResolvedValues
         pendingAnimation?: Process
         currentAnimation?: AnimationPlaybackControls
         mixTargetDelta: (progress: number) => void
+        progress = 0
```
Comment (nvh) [inline packages/framer-motion/src/projection/node/create-projection-node.ts:1097]: I do feel that `animationProgress` is more descriptive, why did you go for `progress` here?

**Review by nvh — APPROVED** [review]

File: packages/framer-motion/src/projection/node/create-projection-node.ts (hunk 3 above)
Comment (mattgperry) [reply packages/framer-motion/src/projection/node/create-projection-node.ts:1097]: Yeah animationProgress makes sense at a class level, I'll change it

**Pull request merged**